### PR TITLE
Integration UI input and LLM API call

### PR DESCRIPTION
**Pull request by adithya1012**
### **Functionality Implemented:**  
- Consumes input from the UI.  
- Converts the input message into a specific format.  
- Instructs the LLM to respond in the designated format using a `SYSTEM_PROMPT`.

### **Output:**  
- The `index.ts` file invokes the LLM with the user's message.  
- The LLM’s output, formatted in XML, is logged to the console.

### **Known Issues:**  
- **TODO:** I have added comments to handle specific edge cases.  
- The LLM response is received in chunks or as a stream. Proper parsing is needed to identify which tool to use.

### **Further Work:**  
- **@Sinduja98** will handle the parsing logic for the streamed content.  
- The next steps include integrating terminal functionalities and invoking the necessary tools.

### CODE REVIEW

**Comment by adithya1012** [comment]
### **Note:**  
To run the program, the Gemini API key needs to be updated.  

- Navigate to:  
  **`DevAssist_AI/src/core/webview/DevAssistProvider.ts`**  
- Update the API key at **line 149**.

### **Action Required:**  
We should modify the implementation to prompt the user to input the API key dynamically.

**Comment by Sinduja98** [comment]
https://aistudio.google.com/app/apikey?_gl=1*58e0ho*_ga*NzU5NjY4MjM2LjE3Mjk5ODIwMDE.*_ga_P1DBVKWT6V*MTcyOTk4MjAwMS4xLjAuMTcyOTk4MjAwMS42MC4wLjE5NTM5NDEyMTM.

Use this URL to create Gemini API if you dont have one.
It's working for me.

**Review by Sinduja98 — APPROVED** [review]

File: src/core/index.ts [hunk 1]
```diff
@@ -19,9 +29,165 @@ export class DevAssist {
...
+					assistantMessage += chunk.text
+					// TODO: parse raw assistant message into content blocks
+					// TODO: present content to user
+					console.log(assistantMessage);
+				}
+			} catch (error) {
+				this.abortTask() 
+			}
+
+			this.didCompleteReadingStream = true
```
Comment (nidhitalastha) [inline src/core/index.ts:96]: Are we planning on streaming the chunks directly to webview UI as and when the extension gets it from the llm or are we planning sending the whole message at once?

Comment (nidhitalastha) [reply src/core/index.ts:96]: I would need the data to be sent to the webview UI in JSON format. as discussed in #3

File: src/core/index.ts [hunk 2]
```diff
@@ -19,9 +29,165 @@ export class DevAssist {
...
+			if (assistantMessage.length > 0) {
+				await this.addToApiConversationHistory({
+					role: "assistant",
+					content: [{ type: "text", text: assistantMessage }],
+				});
+
+				console.log(this.assistantMessageContent);
+				// TODO: Need to identify in the response any of the tool used or not. 
+				// const didToolUse = this.assistantMessageContent.some((block) => block.type === "tool_use")
+				const didToolUse = false;
```
Comment (hdprajwal) [inline src/core/index.ts:110]: Can you confirm if we are not handling tool_use in the current implementation. Are we also planning on sending the the API conversation to the Webview-ui or just the response?

File: src/core/index.ts (hunk 1 above)
Comment (adithya1012) [reply src/core/index.ts:96]: Currently the data is coming in chunks that i am storing in a variable. @Sinduja98 is integrating parsing the each chunk to identify if it is a text or tool_use. For now current implementation we dont need to show data in the UI. 

Once all the data is rendered the content which is wrapped inside the <attempt_complesion> should be sown in UI. Which i will discuss with you.

File: src/core/index.ts (hunk 2 above)
Comment (adithya1012) [reply src/core/index.ts:110]: The current implementation includes tool usage functionalities, such as execute_cmd and write_to_file. However, these operations are currently performed without interacting with the user via the UI for confirmation.

Moving forward, I plan to display the success message to the user. This message will be extracted from the final part of the response, wrapped inside the <attempt_completion> tag.

**Review by jahnavib26 — APPROVED** [review]

**Pull request merged**